---
output: html_document
editor_options: 
  chunk_output_type: console
---


- generate tracks in appropriate format (such as bed/interact.bed)
https://genome.ucsc.edu/goldenPath/help/interact.html

##make bedgraph file for GWAS Manhattan plot

```{r }
library(dplyr)
#make bedgraph for SNPs
# GWAS<-data.table::fread(file = "~/Google Drive/WORK/CF PhD/GWAS/scz2021/PGC3_SCZ_wave3_public.v2_GRCh37.tsv.gz",
#                         select = c("CHR",	"SNP",	"BP", "OR",	"P"), fill=T)
GWAS<-data.table::fread(file = "~/Google Drive/WORK/CF PhD/UCSC_browser/SavageJansen_2018_intelligence_metaanalysis_hg19.txt",
                        select = c("CHR",	"SNP",	"POS", "stdBeta",	"P"), fill=T)
GWAS<-data.table::fread(file = "~/Google Drive/WORK/CF PhD/UCSC_browser/Tadros_GCST011211_hg19.UCSC.bed", fill=T,
                        col.names = c("CHR",	"start",	"end", "SNP",	"P", "delete")) %>% select(-delete) %>%
  mutate(P=(as.numeric(P)))
GWAS<-data.table::fread(file = "~/Google Drive/WORK/CF PhD/GWAS/Sean_2022_HCM.gwama.sumstats", fill=T, 
                        select = c("CHR", "POS", "P", "SNP")) 
GWAS
#GWAS[GWAS$CHR=="1" & GWAS$BP==249220525,]
hist(GWAS$P, breaks = 1000)
GWAS$neglog10p <- -log10(GWAS$P)
hist(GWAS$neglog10p, breaks = 1000)
GWAS[GWAS$P<10^-8,]

# bedgraph format http://genome.ucsc.edu/goldenPath/help/bedgraph.html
# chromA  chromStartA  chromEndA  dataValueA
GWAS$CHR<-paste0("chr",GWAS$CHR)
GWAS$start<-GWAS$POS-1
#GWAS[GWAS$CHR=="chr1" & GWAS$BP==249220525,]
psych::describe(GWAS$neglog10p)
table(GWAS$CHR)
GWAS[GWAS$CHR=="chr23"]$CHR<-"chrX"

#write out header line
cat("track type=bedGraph name=Sean_2022 description=Sean_2022 visibility=display_mode color=125,0,238 altColor=125,0,238 priority=high autoScale=on alwaysZero=on gridDefault=on graphType=bar\n",
    file="/Users/eosimo/Google Drive/WORK/CF PhD/UCSC_browser/Sean_2022_HCM.bedgraph", append=F)
#write out data to same file
data.table::fwrite(
  x = dplyr::select(.data = GWAS, CHR, start, POS, neglog10p) %>% na.exclude(), 
  append = T, sep = "\t", col.names = F, row.names = F, quote = F, scipen = 999,
  file = "/Users/eosimo/Google Drive/WORK/CF PhD/UCSC_browser/Sean_2022_HCM.bedgraph")
```

- make sure bed files are sorted

``` {bash -- bedgraph into bigWig: sort file, remove duplicated lines, convert to wig, upload}
cd /Users/eosimo/Google\ Drive/WORK/CF\ PhD/UCSC_browser/

<!-- sort and remove duplications (-u) based on first 3 cols (-k1,3) only -->
sort -k1,3 -u Sean_2022_HCM.bedgraph | bedtools sort -i > Sean_2022_HCM_sort.bedgraph


bedGraphToBigWig /Users/eosimo/Google\ Drive/WORK/CF\ PhD/UCSC_browser/Sean_2022_HCM_sort.bedgraph http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes /Users/eosimo/Google\ Drive/WORK/CF\ PhD/UCSC_browser/UCSCtrackHub/data_files/Sean_2022_HCM_sort_hg19.bw

<!-- copy onto shire -->
rsync -P /Users/eosimo/Google\ Drive/WORK/CF\ PhD/UCSC_browser/UCSCtrackHub/data_files/Sean_2022_HCM_sort_hg19.bw emanuele@193.60.222.200:/mnt/storage/www/ucsc_hub/eosimo/
```


##make bigBed file for e.g. LD blocks and other genomic features such as SNVs

```{r make bed from csv}
library(tidyverse)
library(GenomicRanges)
library(rtracklayer)

feature <- data.table::fread("/Users/eosimo/Google Drive/WORK/CF PhD/UCSC_browser/May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg38.csv", select = c("seqnames",    "start",      "end", "scaled_ES", "variant_id", "enh", "ensembl_gene_id")) %>% 
  tidyr::unite(name, c("variant_id", "enh", "ensembl_gene_id")) %>% dplyr::rename(score=scaled_ES) %>% 
  mutate(score=round(scales::rescale(score, to = c(0,1000)),0)) %>% #select(-variant) %>% 
  makeGRangesFromDataFrame(
  start.field = "start",  end.field = "end", seqnames.field = "seqnames", ignore.strand = T,
  keep.extra.columns = T)
seqlevelsStyle(feature) = "UCSC"

#convert to hg19 if necessary
ch = import.chain("/Users/eosimo/Google Drive/WORK/CF PhD/UCSC_browser/hg38ToHg19.over.chain")
#The liftOver function will create a GRangesList.
feature_hg19 <- liftOver(feature, ch) %>%  unlist() %>% 
  as_tibble() %>% unique() %>%  
  makeGRangesFromDataFrame(keep.extra.columns = T)
feature_hg19

#need to call name and score the fields to appear in BED file:
export.bed(format = "bed",  object=feature_hg19,con="/Users/eosimo/Google Drive/WORK/CF PhD/UCSC_browser/May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg19.bed" )

```

``` {bash -- bed into bigBed: sort file, remove duplicated lines, convert to bb, upload}
cd /Users/eosimo/Google\ Drive/WORK/CF\ PhD/UCSC_browser/

<!-- sort and remove duplications (-u) based on first 3 cols (-k1,3) only -->
sort -k1,3 -u May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg19.bed | bedtools sort -i > May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg19_sort.bed

<!-- simple bed file -->
bedToBigBed -type=bed6 May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg19_sort.bed http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg19_sort.bb

#If you add extra fields to your bigBed file, you must include an AutoSql format (.as) file describing the fields. In this file, all fields (standard and non-standard) are described with a short internal name and also a human-readable description. For more information on AutoSql, see Kent and Brumbaugh, 2002, as well as examples of .as files in this directory. https://genome.ucsc.edu/goldenPath/help/bigBed.html
bedToBigBed -as=bb.as -type=bed6+4 brainSNVs_hg19_UCSC_sort.bed http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes brainSNVs_hg19_UCSC_sort.bb



<!-- copy onto shire -->
rsync -P May22_significant_SNVs_allEPpairs_ESpos_someContact_allTotScores_anyTissue_overlapeQTLsMeta_perSNV_p0.05_meta_BRAIN_hg19_sort.bb emanuele@193.60.222.200:/mnt/storage/www/ucsc_hub/eosimo/
```

the link to load the track will be trackhub.genereg.net/eosimo/



